cmake/configure/configure_20_trilinos.cmake

```diff
@@ -272,9 +272,45 @@ macro(feature_trilinos_find_external var)
       else()
         message(
           STATUS
-          "Tpetra was found but is not usable! Disabling Tpetra support."
+          "Tpetra was found but is not usable! Disabling Tpetra support and searching for common errors."
           )
         set(TRILINOS_WITH_TPETRA OFF)
+
+        check_cxx_symbol_exists(HAVE_TPETRA_INT_LONG_LONG "TpetraCore_config.h" DEAL_II_HAVE_TPETRA_INT_LONG_LONG)
+        check_cxx_symbol_exists(HAVE_TPETRA_INT_INT "TpetraCore_config.h" DEAL_II_HAVE_TPETRA_INT_INT)
+
+        if(DEAL_II_HAVE_TPETRA_INT_INT AND DEAL_II_WITH_64BIT_INDICES)
+          message(
+            STATUS 
+            "  Tpetra is installed with 32 bit global indices but you want to build deal.II with 64. "
+          )
+          message(
+            STATUS 
+            "  Either rebuild Trilinos with -DTPETRA_INST_INT_INT=OFF and -DTPETRA_INST_LONG_LONG=ON "
+          )
+          message(
+            STATUS 
+            "  or clean and reconfigure deal.II with -DDEAL_II_WITH_64_BIT_INDICES=OFF."
+          )
+        elseif(DEAL_II_HAVE_TPETRA_INT_LONG_LONG AND NOT DEAL_II_WITH_64BIT_INDICES)
+          message(
+            STATUS 
+            "  Tpetra is installed with 64 bit global indices but you want to build deal.II with 32. "
```

Comment:
```suggestion
            "  Tpetra doesn't support 32-bit global indices but deal.II is configured to use 32-bit global indices. "
```

Reply:
'doesn't support' makes it sound like it's not a valid option in my opinion.
How about
'was configured with'?

Reply:
Sure, we could do "The Tpetra configuration doesn't support [...]", "Tpetra wasn't configured with support for [...]" or something like that. I don't care very much about the exact words used. 🙂

Reply:
Fixed it to the second option.